webpack/assets/javascripts/react_app/routes/RegistrationCommands/RegistrationCommandsPage/components/fields/RepositoryModalInput.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+
+import PropTypes from 'prop-types';
+
+import { FormGroup, TextInput, Grid } from '@patternfly/react-core';
+
+const RepositoryModalInput = ({
+  reference,
+  updateRepository,
+  isLoading,
+  initRepo,
+  initGpgKey,
+}) => {
+  const onRepoUpdate = repo => {
+    updateRepository(reference, repo, 'repository');
+  };
+
+  const onGpgKeyUpdate = gpgKey => {
+    updateRepository(reference, gpgKey, 'gpgKeyUrl');
+  };
+
+  return (
+    <Grid hasGutter md={6}>
+      <FormGroup fieldId="reg_repo">
+        <TextInput
+          ouiaId="reg_host_repo"
+          id="reg_host_repo"
+          key="reg_host_repo"
+          value={initRepo}
+          type="text"
```

Comment:
Should it have a check that user adds correct format of the repository? Right now, I can add anything and it accepts it.

Reply:
True. We could add something like this later on. A check would require the OS to be set as the format differs for different operating systems. And the OS field is not even mandatory. In the current implementation, you can also enter anything and it would just accept it.